### ISSUE TITLE
Crash

### ISSUE THREAD

**Issue by blake-sc** [nevir/oni-darkness-not-excluded#2]
No obvious gameplay cause.

```
Error in Rowan.Light2D.OnSpawn 
System.IndexOutOfRangeException: Index was outside the bounds of the array.
  at DarknessNotIncluded.Exploration.LightsRevealFogOfWar+Patched_LightGridManager_LightGridEmitter_AddToGrid.Postfix (System.Collections.Generic.List`1[T] ___litCells) [0x00013] in <53ae20cf2ee74c2f972a133dd7141340>:0 
  at (wrapper dynamic-method) LightGridManager+LightGridEmitter.LightGridManager+LightGridEmitter.AddToGrid_Patch1(LightGridManager/LightGridEmitter,bool)
  at LightGridManager+LightGridEmitter.Refresh (LightGridManager+LightGridEmitter+State state, System.Boolean force) [0x00025] in <b93054555ee347b19133a9b09757cab5>:0 
  at Light2D.OnSpawn () [0x00040] in <b93054555ee347b19133a9b09757cab5>:0 
  at KMonoBehaviour.Spawn () [0x0005d] in <773696405cb3498b823b04c854905869>:0 

  at UnityEngine.Debug.LogError (System.Object message, UnityEngine.Object context) [0x00000] in <a83e096d0ba5413da59ce235b648c30c>:0 
  at Debug.LogError (System.Object obj, UnityEngine.Object context) [0x00000] in <773696405cb3498b823b04c854905869>:0 
  at DebugUtil.LogErrorArgs (UnityEngine.Object context, System.Object[] objs) [0x00000] in <773696405cb3498b823b04c854905869>:0 
  at DebugUtil.LogException (UnityEngine.Object context, System.String errorMessage, System.Exception e) [0x00000] in <773696405cb3498b823b04c854905869>:0 
  at KMonoBehaviour.Spawn () [0x00000] in <773696405cb3498b823b04c854905869>:0 
  at Util.SpawnComponent (UnityEngine.Component cmp) [0x00000] in <773696405cb3498b823b04c854905869>:0 
  at MyAttributeManager`1[T].OnStart (System.Object obj, KMonoBehaviour cmp) [0x00000] in <773696405cb3498b823b04c854905869>:0 
  at MyAttributes.OnStart (System.Object obj, KMonoBehaviour cmp) [0x00000] in <773696405cb3498b823b04c854905869>:0 
  at StateMachine+Instance.StartSM () [0x00000] in <b93054555ee347b19133a9b09757cab5>:0 
  at StateMachine`4+GenericInstance[StateMachineType,StateMachineInstanceType,MasterType,DefType].StartSM () [0x00000] in <b93054555ee347b19133a9b09757cab5>:0 
  at GlowStick.OnSpawn () [0x00000] in <b93054555ee347b19133a9b09757cab5>:0 
  at KMonoBehaviour.Spawn () [0x00000] in <773696405cb3498b823b04c854905869>:0 
  at KMonoBehaviour.Start () [0x00000] in <773696405cb3498b823b04c854905869>:0 
Build: U46-552078-S
```

**Comment by extra-salad** [comment]
This happened to me once after I rejected the printing pod.

**Comment by gitshit** [comment]
Same issue when rerolling dupes on colony start

**Comment by nevir** [comment]
Sorry for the crash folks! Hoping to address it in the next day or two

**Comment by Taoquitok** [comment]
@nevir, FYI for your upcoming patch, had the same crash error during the character select screen prior to loading a new game

```log
Error in Stinky.Light2D.OnSpawn 
System.IndexOutOfRangeException: Index was outside the bounds of the array.
  at DarknessNotIncluded.Exploration.LightsRevealFogOfWar+Patched_LightGridManager_LightGridEmitter_AddToGrid.Postfix (System.Collections.Generic.List`1[T] ___litCells) [0x00013] in <53ae20cf2ee74c2f972a133dd7141340>:0 
  at (wrapper dynamic-method) LightGridManager+LightGridEmitter.LightGridManager+LightGridEmitter.AddToGrid_Patch1(LightGridManager/LightGridEmitter,bool)
  at LightGridManager+LightGridEmitter.Refresh (LightGridManager+LightGridEmitter+State state, System.Boolean force) [0x00025] in <1f4c5a7d658c4522888fade7e74323ad>:0 
  at Light2D.OnSpawn () [0x00040] in <1f4c5a7d658c4522888fade7e74323ad>:0 
  at KMonoBehaviour.Spawn () [0x0005d] in <d35824e72985464e85149179dd14d71e>:0 

  at UnityEngine.Debug.LogError (System.Object message, UnityEngine.Object context) [0x00000] in <72b60a3dd8cd4f12a155b761a1af9144>:0 
  at Debug.LogError (System.Object obj, UnityEngine.Object context) [0x00000] in <d35824e72985464e85149179dd14d71e>:0 
  at DebugUtil.LogErrorArgs (UnityEngine.Object context, System.Object[] objs) [0x00000] in <d35824e72985464e85149179dd14d71e>:0 
  at DebugUtil.LogException (UnityEngine.Object context, System.String errorMessage, System.Exception e) [0x00000] in <d35824e72985464e85149179dd14d71e>:0 
  at KMonoBehaviour.Spawn () [0x00000] in <d35824e72985464e85149179dd14d71e>:0 
  at Util.SpawnComponent (UnityEngine.Component cmp) [0x00000] in <d35824e72985464e85149179dd14d71e>:0 
  at MyAttributeManager`1[T].OnStart (System.Object obj, KMonoBehaviour cmp) [0x00000] in <d35824e72985464e85149179dd14d71e>:0 
  at MyAttributes.OnStart (System.Object obj, KMonoBehaviour cmp) [0x00000] in <d35824e72985464e85149179dd14d71e>:0 
  at StateMachine+Instance.StartSM () [0x00000] in <1f4c5a7d658c4522888fade7e74323ad>:0 
  at StateMachine`4+GenericInstance[StateMachineType,StateMachineInstanceType,MasterType,DefType].StartSM () [0x00000] in <1f4c5a7d658c4522888fade7e74323ad>:0 
  at GlowStick.OnSpawn () [0x00000] in <1f4c5a7d658c4522888fade7e74323ad>:0 
  at KMonoBehaviour.Spawn () [0x00000] in <d35824e72985464e85149179dd14d71e>:0 
  at KMonoBehaviour.Start () [0x00000] in <d35824e72985464e85149179dd14d71e>:0 
Build: U46-552078-SD
```

**Comment by yandayixia** [comment]
Can confirm this happens regularly at loading care package page, reject all options selection, and when refreshing pawn selection at the start of a new game. Not reproduceable 100% as upon reloading there's a good chance the crash won't happen.

**Comment by nevir** [comment]
I believe I have this fixed in 1.2.0 - please update (and use [Mod Updater](https://steamcommunity.com/sharedfiles/filedetails/?id=2018291283) if you don't have it) and let me know if you're still experiencing it!
